core/src/test/java/io/confluent/rest/SslTest.java

```diff
@@ -60,28 +63,49 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertTrue;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.fail;
 
 public class SslTest {
   private static final Logger log = LoggerFactory.getLogger(SslTest.class);
 
-  private File trustStore;
-  private File clientKeystore;
-  private File serverKeystore;
-  private File serverKeystoreBak;
-  private File serverKeystoreErr;
+  private static File trustStore;
+  private static File clientKeystore;
+  private static File serverKeystore;
+  private static File serverKeystoreBak;
+  private static File serverKeystoreErr;
 
   public static final String SSL_PASSWORD = "test1234";
   public static final String EXPECTED_200_MSG = "Response status must be 200.";
-  public static final int CERT_RELOAD_WAIT_TIME = 30000;
-
-  @Before
-  public void setUp() throws Exception {
+  public static final int CERT_RELOAD_WAIT_TIME = 20000;
+
+  private static TemporaryFolder tempFolder;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+
+    /*
+     * To make this test less flakey
+     *  - 1 don't create keystore files for every test method
+     *  - 2 cleanup keystore files on test exist so they don't have to be considerd by the FileWatcher
+     *  - 3 updated the FileWatcher and Application class to not have a single shared threadpool to
+     *      watch for changed files.
+     *
+     * By default temp files are not cleaned when up.  Which isn't normally a problem unless you are
+     * testing the ability of rest-utils apps to notice and reload updated ssl keystore files.
+     *
+     * Turns out the temp dir that Java+MacOs was continually using on my local machine had 1500
+     * files in it.  Also the Java "FileWatcher" for Mac works via polling a directory,
+     * this seems to have added to the flakeyness.
+     */
+    tempFolder = new TemporaryFolder();
+    tempFolder.create();
     try {
-      trustStore = File.createTempFile("SslTest-truststore", ".jks");
-      clientKeystore = File.createTempFile("SslTest-client-keystore", ".jks");
-      serverKeystore = File.createTempFile("SslTest-server-keystore", ".jks");
-      serverKeystoreBak = File.createTempFile("SslTest-server-keystore", ".jks.bak");
-      serverKeystoreErr = File.createTempFile("SslTest-server-keystore", ".jks.err");
+      trustStore = File.createTempFile("SslTest-truststore", ".jks", tempFolder.getRoot());
```

Comment:
Any reason to prefer this to `tempFolder.newFile`?

Reply:
No.